storage/innobase/fil/fil0fil.cc

```diff
@@ -6893,6 +6893,9 @@ bool Fil_shard::space_extend(fil_space_t *space, page_no_t size) {
       log_write_up_to() is not needed here, because no file shrinks and
       duplicate extending is allowed. And log_write_up_to() here helps
       nothing for fallocate() inconsistency. */
+#if defined(XTRABACKUP)
+      log_write_up_to(*log_sys, mtr.commit_lsn(), true);
+#endif /* XTRABACKUP */
```

Comment:
When you have time, can you please check why the pages in flush_list are not flushed without this change?

Guesses:
1. Pages in flush list do not have proper LSN set?
2. are we creating pages without writing redo?

Reply:
As per slack conversation, the issue comes when we call `fil_op_write_space_extend` this is generating the extra redo that causes the hang at shutdown. We will skip the whole block for pxb.